Smarty/Smarty_Compiler.class.php

```diff
@@ -556,7 +551,7 @@ function _compile_tag($template_tag)
 
             case 'php':
                 /* handle folded tags replaced by {php} */
-                list(, $block) = each($this->_folded_blocks);
+                $block = array_shift($this->_folded_blocks);
```

Comment:
Most of the changes look like clear equivalents. The loop through `$this->_folded_blocks` is the only one which feels like it *could potentially* have side-effects (depending on how `$this->_folded_blocks` is used elsewhere), but I'd defer to upstream if they like it.